File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (n) => {

  let output = 0;

  if(n < 1){
    throw new Error("Only positive numbers are allowed")
  }

  while(n != 1){

    if(n % 2 == 0){
      n /= 2;
    }
    else{
      n = 3 * n + 1;
    }

    output++;

  }

  return output;

};

File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):

    # error handling
    if number <= 0:
        raise ValueError("Has to be a positive integer")

    out = 0

    while number != 1:
        out += 1

        if(number % 2 == 0):
            number /= 2
        else:
            number = 3*number + 1

    return out